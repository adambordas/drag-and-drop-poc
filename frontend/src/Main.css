.container {
  width: 800px;
  display: flex;
  margin: auto;
}
