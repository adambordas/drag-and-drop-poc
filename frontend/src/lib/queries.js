import { gql } from '@apollo/client';

export const GET_CARDS = gql`
  query GetCards {
    cards {
      id,
      description,
      lane
    }
  }
`;

export const MOVE_CARD = gql`
  mutation UpdateCard($id: ID!, $lane: String!) {
    updateCard(id: $id, lane: $lane) {
      id,
      description,
      lane
    }
  }
`;
