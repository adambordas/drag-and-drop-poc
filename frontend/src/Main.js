import './Main.css';
import Lane from './components/lane/Lane'
import { useEffect, useState } from 'react';
import { useQuery } from '@apollo/client';
import { GET_CARDS } from './lib/queries';

function Main() {
  const [cards, setCards] = useState([]);

  const { data } = useQuery(GET_CARDS, { fetchPolicy: 'network-only' });

  useEffect(() => {
    if (data) {
      setCards(data.cards);
    }
  }, [data]);

  return (
    <div className="container">
      <Lane
        name="todo"
        cards={cards.filter(card => card.lane === 'todo')}
      />
      <Lane
        name="in-progress"
        cards={cards.filter(card => card.lane === 'in-progress')}
      />
      <Lane
        name="done"
        cards={cards.filter(card => card.lane === 'done')}
      />
    </div>
  );
}

export default Main;
