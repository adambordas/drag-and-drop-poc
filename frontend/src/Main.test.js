import React from 'react';
import { act, cleanup, render } from '@testing-library/react';
import Main from './Main';
import { MockedProvider } from '@apollo/client/testing';
import { GET_CARDS } from './lib/queries';

const cardsData = [{
  id: 'test id 1',
  description: 'test description 1',
  lane: 'todo',
}, {
  id: 'test id 2',
  description: 'test description 2',
  lane: 'in-progress',
}, {
  id: 'test id 3',
  description: 'test description 3',
  lane: 'done',
}];

describe('Main component', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders properly', async () => {
    const mocks = [
      {
        request: {
          query: GET_CARDS,
        },
        result: {
          data: {
            cards: cardsData
          },
        },
      },
    ];

    const { container, getByText, queryAllByText } = render(
      <MockedProvider mocks={mocks}>
        <Main />
      </MockedProvider>
    );
    
    await act(async () => {
      await new Promise(resolve => setTimeout(resolve, 0));
    });

    const todoTitle = getByText('todo');
    const inProgressTitle = getByText('in-progress');
    const doneTitle = getByText('done');
    const cards = queryAllByText(/^test description/);

    expect(todoTitle).toBeInTheDocument();
    expect(inProgressTitle).toBeInTheDocument();
    expect(doneTitle).toBeInTheDocument();
    expect(cards).toHaveLength(cardsData.length);
    expect(container).toMatchSnapshot();
  });
});
