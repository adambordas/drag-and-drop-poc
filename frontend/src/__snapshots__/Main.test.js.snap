// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders properly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="lane-container"
    >
      <h1>
        todo
      </h1>
      <div
        class="lane"
      >
        <div
          class="card"
          draggable="true"
        >
          test description 1
        </div>
      </div>
    </div>
    <div
      class="lane-container"
    >
      <h1>
        in-progress
      </h1>
      <div
        class="lane"
      >
        <div
          class="card"
          draggable="true"
        >
          test description 2
        </div>
      </div>
    </div>
    <div
      class="lane-container"
    >
      <h1>
        done
      </h1>
      <div
        class="lane"
      >
        <div
          class="card"
          draggable="true"
        >
          test description 3
        </div>
      </div>
    </div>
  </div>
</div>
`;
